index.js

```diff
@@ -1,11 +1,29 @@
+const fs = require('node:fs')
+const path = require('node:path')
+
+const LIST_ERRORS = {
+  NO_PATH: 'No se ha especificado el path del archivo',
+  NO_WORD: 'No se ha especificado la palabra a buscar'
+}
 // Ejercicio 2
 async function writeFile (filePath, data, callback) {
-
+  const dirPath = path.dirname(filePath);
+  fs.mkdirSync(dirPath, { recursive: true });
+  fs.writeFile(filePath, data, callback);
 }
-
 // Ejercicio 3
-async function readFileAndCount (word, callback) {
-
+async function readFileAndCount(word, callback) {
+  const argv = process.argv[2];
+  if (!argv) return callback(new Error(LIST_ERRORS.NO_PATH)); // r: No se ha especificado el path del archivo
+  if (!word) return callback(new Error(LIST_ERRORS.NO_WORD)); // r: No se ha especificado la palabra a buscar
+  fs.exists(argv, (existFilePath) => {
```

Comment:
```suggestion
  const argv = process.argv[2];
  
  if (!argv) return callback(new Error(LIST_ERRORS.NO_PATH)); // r: No se ha especificado el path del archivo
  if (!word) return callback(new Error(LIST_ERRORS.NO_WORD)); // r: No se ha especificado la palabra a buscar
  
  fs.exists(argv, (existFilePath) => {
```

Para poder seguir mejor la cadencia del código, es buena idea separar con espacios cada cosa.

La creación de variables, la validación y finalmente la lógica.

Reply:
Súper! Gracias por el feedback Midu, mejorare y actualizaré estas observaciones.

Me gustó cada detalle de los ejercicios ❤️

Reply:
Gracias, crack!